omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,133 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class IdentifierSourceRestControllerTest extends MainResourceControllerTest {
+	
+	public static final String IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a999";
+	
+	private IdentifierSourceService service;
+
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestDataForLoadTest.xml");
+		this.service = Context.getService(IdentifierSourceService.class);
+	}
+
+	@Override
+	public String getURI() {
+		return "idgen/identifiersource";
+	}
+
+	@Override
+	public String getUuid() {
+		return IDENTIFIER_SOURCE_UUID;
+	}
+
+	@Override
+	public long getAllCount()  {
+		return service.getAllIdentifierSources(false).size();
+	}
+	
+	@Test
+	public void shouldEditAnIdentifierSource() throws Exception {
+		
+		String description = "{\"description\":\"Updated description\"}";
+		
+		handle(newPostRequest(getURI() + "/" + getUuid(), description));
+		
+		assertEquals("Updated description", service.getIdentifierSourceByUuid(getUuid()).getDescription());
+		
+	}
+	
+	@Test
+	public void shouldVoidAnIdentifierSource() throws Exception {
+		
+		final String reason = "none";
+		
+		assertEquals(false, service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", reason);
+		handle(req);
+		
+		assertEquals(true, service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		assertEquals(reason, service.getIdentifierSourceByUuid(getUuid()).getRetireReason());
+
+	}
+	
+	@Test
+	public void shouldPurgeAnIdentifierSource() throws Exception {
+		
+		assertNotNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("purge", "")));
+		
+		assertNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+	}	
+	
+	@Test
+	public void shouldGetAll() throws Exception  {
+		
+		SimpleObject result = deserialize(handle(newGetRequest(getURI())));
+			
+		assertEquals(getAllCount(), Util.getResultsSize(result));
+		
+	}
+
+	@Test
+	public void shouldGetAnIdentifierSourceByUuid() throws Exception {
+		
+		SimpleObject result = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+				
+		assertNotNull(result);
+		assertEquals(getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	
+	}
+
+	@Test
+	public void shouldSaveIdentifierSource() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+			
+		SimpleObject identifierSource = new SimpleObject();
+		identifierSource.add("name", "test identifier source");
+		identifierSource.add("description", "test identifier source");
+		identifierSource.add("identifierType", Context.getPatientService().getPatientIdentifierType(1));
+				
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
```

Comment:
To get rid of the identifierType.retired error, can you remove new ObjectMapper().writeValueAsString and replace it by building the json as in this class? https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ProviderController1_9Test.java

Reply:
fixed